src/main/scala/is/hail/sparkextras/ContextRDD.scala

```diff
@@ -165,6 +170,27 @@ class ContextRDD[C <: AutoCloseable, T: ClassTag](
     else
       onRDD(_.coalesce(numPartitions, shuffle))
 
+  def sample(
+    withReplacement: Boolean,
+    fraction: Double,
+    seed: Long
+  ): ContextRDD[C, T] = {
+    require(fraction >= 0.0 && fraction <= 1.0)
+    val r = new Random(seed)
+    val partitionSeeds =
+      sparkContext.broadcast(
+        Array.tabulate(rdd.partitions.length)(_ => r.nextLong()))
+    val makeSampler = if (withReplacement)
+      () => new PoissonSampler[T](fraction)
+    else
+      () => new BernoulliSampler[T](fraction)
+    cmapPartitionsWithIndex({ (i, ctx, it) =>
+      val sampler = makeSampler()
```

Comment:
Why not just do the if in here?

Reply:
🤷‍♀️ seemed like it should be done once. I'll push it in.